File: php/110-balanced-binary-tree.php
<?php
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     public $val = null;
 *     public $left = null;
 *     public $right = null;
 *     function __construct($value) { $this->val = $value; }
 * }
 */
class Solution 
{
    /**
     * @param TreeNode $root
     * @return Boolean
     */
    public function isBalanced($root) 
    {
        if ($root == null) {
            return true;
        }
        if (abs($this->maxDepth($root->left)-$this->maxDepth($root->right)) > 1) {
            return false;
        }
        return $this->isBalanced($root->left) && $this->isBalanced($root->right);   
    }
    public function maxDepth($node) 
    {
        if ($node == null) {
            return 0;
        }
        return max($this->maxDepth($node->left), $this->maxDepth($node->right)) + 1;
    }
}

File: go/110-balanced-binary-tree.go
package main

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func isBalanced(root *TreeNode) bool {
	if root == nil {
		return true
	}
	leftMax := maxDepth(root.Left)
	rightMax := maxDepth(root.Right)
	if int(math.Abs(leftMax-rightMax))-1 > 0 {
		return false
	}
	return isBalanced(root.Left) && isBalanced(root.Right)
}
func maxDepth(root *TreeNode) float64 {
	if root == nil {
		return float64(0)
	}
	return math.Max(float64(maxDepth(root.Left)), float64(maxDepth(root.Right))) + 1
}
